### DISCUSSION TITLE
No agent  class hierarchy

### DISCUSSION THREAD

**Discussion by rhohndorf** [Maximilian-Winter/llama-cpp-agent#24] [Q&A]
Although LlamaCppAgent is described as a base agent none of the other agent classes inherits from it. Are there plans to refactor/restructure this ?

**Comment by Maximilian-Winter** [comment]
@rhohndorf Yes, I want to refactor this. Thank you for your support with the pull requests. I will try to do this next weekend and add some other things and update the documentation. I want to release a new version with some updates.
